### ISSUE TITLE
Using multiple db contexts with SQlite in-memory tests results in wrong result

### ISSUE THREAD

**Issue by StarpTech** [aspnet/EntityFramework6#299]
I use Sqlite in-memory database to write tests for my application as described here and in the docs https://github.com/aspnet/EntityFramework.Docs/blob/master/samples/core/Miscellaneous/Testing/TestProject/SQLite/BlogServiceTests.cs

But my tests fails often when I open for every section a new database context with
```c#
// In-memory database only exists while the connection is open
var connection = new SqliteConnection("DataSource=:memory:");
connection.Open();

try
{
  var options = new DbContextOptionsBuilder<BloggingContext>()
      .UseSqlite(connection)
      .Options;
      
  // Save items
  using (var context = new BloggingContext(options))
  { 
      .....
  }
  // Use a separate instance of the context to verify correct data was saved to database
  using (var context = new BloggingContext(options))
  { 
      .....
  }

}
finally
{
    connection.Close();
}
```
when I collapse the propertys manually in the debug mode the data appears suddenly or sometimes not. When I don't using the seperate db context everything works fine.

**Comment by StarpTech** [comment]
In this variation the data is present in the second db context. I hope it helps.

```c#
// In-memory database only exists while the connection is open
var connection = new SqliteConnection("DataSource=:memory:");
connection.Open();

try
{
  var options = new DbContextOptionsBuilder<BloggingContext>()
      .UseSqlite(connection)
      .Options;
      
  // Save items
  .....

  // Use a separate instance of the context to verify correct data was saved to database
  using (var context = new BloggingContext(options))
  { 
      .....
  }

}
finally
{
    connection.Close();
}
```

**Comment by ajcvickers** [comment]
@StarpTech This looks like an issue that was fixed in EF Core 1.1.2. If you are already using 1.1.2 (or a pre-release of 2.0.0) then please open an issue on https://github.com/aspnet/EntityFramework. (This github repo is for EF6 issues.)
